lib/permutations.rb

```diff
@@ -1,4 +1,25 @@
+#Write a method which will take two strings as arguments and returns true if one string is a permutation of the other.
+#Time complexity: O(n+m)
+#Space complexity: O(n)
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
-end
\ No newline at end of file
+  return false if string1.length != string2.length
+ 
+  hash = Hash.new
+
+  string1.chars.each do |char|
+    hash[char] = 1
+  end
```

Comment:
What about the numbers of each letter.  Your solution will indicate that `heelo` and `hello` are permutations of each other.